src/main/java/evelyn/command/TaskList.java

```diff
@@ -0,0 +1,216 @@
+package evelyn.command;
+
+import evelyn.task.Deadline;
+import evelyn.task.Event;
+import evelyn.task.Task;
+import evelyn.task.Todo;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+
+import java.util.ArrayList;
+
+/**
+ * Houses all the logic for a Tasklist in the chatbot.
+ */
+public class TaskList {
+    private Storage storage;
+    private ArrayList<Task> list;
+
+    /**
+     * Constructor of a TaskList object.
+     * @param storage Storage used for the chatbot.
+     */
+    public TaskList(Storage storage) {
+        this.list = new ArrayList(100);
+        try {
+            this.storage = storage;
+            if (!this.storage.fileExists()) {
+                this.storage.createNewDataFile();
+            } else {
+                BufferedReader br = new BufferedReader(this.storage.getFileReader());
+                String line;
+                while ((line = br.readLine()) != null) {
+                    fileDataToList(line);
+                }
+            }
+        } catch (IOException e) {
+            System.err.println("Error creating new file / restoring file data: " + e);
+        }
+    }
+
+    /**
+     * Lists all the tasks that have been input.
+     */
+    public void listTask() {
+        System.out.println(Ui.horizontalLine);
+        System.out.println("Here are the tasks in your list: ");
+        for (int i = 0; i < this.listSize(); i++) {
+            System.out.println((i + 1) + "." + this.getTask(i).toString());
+        }
+        System.out.println(Ui.horizontalLine);
+    }
+
+    /**
+     * Removes a task from the list at a particular index.
+     * @param index Index of the task
+     */
+    public void removeTask(int index) {
+        Task task = (Task) list.get(index);
+        list.remove(index);
+        System.out.println(Ui.horizontalLine);
+        System.out.println("Noted. I've removed this task:");
+        System.out.println("   " + task.toString());
+        System.out.println(this.listSize() > 1 ? "Now you have " + this.listSize() + " tasks in this list"
+                : "Now you have " + this.listSize() + " task in this list");
+        System.out.println(Ui.horizontalLine);
+    }
+
+    /**
+     * Adds a task to the list.
+     * @param task Task to be added to the list.
+     */
+    public void addTask(Task task) {
+        try {
+            list.add(task);
+            storage.writeToFile(task);
+            System.out.println(Ui.horizontalLine);
+            System.out.println("Got it. I've added this task:");
+            System.out.println("  " + task.toString());
+            System.out.println(this.listSize() > 1 ? "Now you have " + this.listSize() + " tasks in this list"
```

Comment:
Could shift printing number of tasks into a method of its own, allows for easier maintenance!